ccl/streamingccl/streamingest: TestTenantStreamingUnavailableStreamAddress failed

ccl/streamingccl/streamingest.TestTenantStreamingUnavailableStreamAddress [failed](https://tanzanite.cluster.engflow.com/invocation/817e8f2d-e4dc-4b32-bfbb-783ac3877f53) on master @ [59a69ae0c5474a8b323afac269055bf6ce8afa34](https://github.com/cockroachdb/cockroach/commits/59a69ae0c5474a8b323afac269055bf6ce8afa34):


```
=== RUN   TestTenantStreamingUnavailableStreamAddress
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantStreamingUnavailableStreamAddress3357960898
    test_log_scope.go:81: use -show-logs to present logs inline

pkg/ccl/streamingccl/replicationtestutils/testutils.go:149: ((*TenantStreamingClusters).init)
	NOTICE: .WaitForTenantCapabilities() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().WaitForTenantCapabilities() instead.
    testutils.go:487: condition failed to evaluate within 45s: from testutils.go:495: stream ingestion has not recorded any progress yet, waiting to advance pos 1697825239.858780119,0
    panic.go:522: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantStreamingUnavailableStreamAddress3357960898
--- FAIL: TestTenantStreamingUnavailableStreamAddress (378.50s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=26</code>
, <code>shard=9</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStreamingUnavailableStreamAddress.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32592

ccl/streamingccl/streamingest.TestTenantStreamingUnavailableStreamAddress [failed](https://tanzanite.cluster.engflow.com/invocation/817e8f2d-e4dc-4b32-bfbb-783ac3877f53) on master @ [59a69ae0c5474a8b323afac269055bf6ce8afa34](https://github.com/cockroachdb/cockroach/commits/59a69ae0c5474a8b323afac269055bf6ce8afa34):


```
=== RUN   TestTenantStreamingUnavailableStreamAddress
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantStreamingUnavailableStreamAddress4030900305
    test_log_scope.go:81: use -show-logs to present logs inline

pkg/ccl/streamingccl/replicationtestutils/testutils.go:149: ((*TenantStreamingClusters).init)
	NOTICE: .WaitForTenantCapabilities() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().WaitForTenantCapabilities() instead.
    testutils.go:487: condition failed to evaluate within 45s: from testutils.go:495: stream ingestion has not recorded any progress yet, waiting to advance pos 1697824761.999885085,0
    panic.go:522: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantStreamingUnavailableStreamAddress4030900305
--- FAIL: TestTenantStreamingUnavailableStreamAddress (404.74s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=16</code>
, <code>shard=9</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStreamingUnavailableStreamAddress.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/streamingccl/streamingest.TestTenantStreamingUnavailableStreamAddress [failed](https://tanzanite.cluster.engflow.com/invocation/817e8f2d-e4dc-4b32-bfbb-783ac3877f53) on master @ [59a69ae0c5474a8b323afac269055bf6ce8afa34](https://github.com/cockroachdb/cockroach/commits/59a69ae0c5474a8b323afac269055bf6ce8afa34):


```
=== RUN   TestTenantStreamingUnavailableStreamAddress
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantStreamingUnavailableStreamAddress710695333
    test_log_scope.go:81: use -show-logs to present logs inline

pkg/ccl/streamingccl/replicationtestutils/testutils.go:149: ((*TenantStreamingClusters).init)
	NOTICE: .WaitForTenantCapabilities() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().WaitForTenantCapabilities() instead.
    testutils.go:487: condition failed to evaluate within 45s: from testutils.go:495: stream ingestion has not recorded any progress yet, waiting to advance pos 1697824032.455989410,0
    panic.go:522: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantStreamingUnavailableStreamAddress710695333
--- FAIL: TestTenantStreamingUnavailableStreamAddress (385.18s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=5</code>
, <code>shard=9</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStreamingUnavailableStreamAddress.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/streamingccl/streamingest.TestTenantStreamingUnavailableStreamAddress [failed](https://tanzanite.cluster.engflow.com/invocation/817e8f2d-e4dc-4b32-bfbb-783ac3877f53) on master @ [59a69ae0c5474a8b323afac269055bf6ce8afa34](https://github.com/cockroachdb/cockroach/commits/59a69ae0c5474a8b323afac269055bf6ce8afa34):


```
=== RUN   TestTenantStreamingUnavailableStreamAddress
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantStreamingUnavailableStreamAddress3820995876
    test_log_scope.go:81: use -show-logs to present logs inline

pkg/ccl/streamingccl/replicationtestutils/testutils.go:149: ((*TenantStreamingClusters).init)
	NOTICE: .WaitForTenantCapabilities() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().WaitForTenantCapabilities() instead.
    testutils.go:487: condition failed to evaluate within 45s: from testutils.go:495: stream ingestion has not recorded any progress yet, waiting to advance pos 1697825245.831259974,0
    panic.go:522: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantStreamingUnavailableStreamAddress3820995876
--- FAIL: TestTenantStreamingUnavailableStreamAddress (333.16s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=30</code>
, <code>shard=9</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStreamingUnavailableStreamAddress.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/streamingccl/streamingest.TestTenantStreamingUnavailableStreamAddress [failed](https://tanzanite.cluster.engflow.com/invocation/817e8f2d-e4dc-4b32-bfbb-783ac3877f53) on master @ [59a69ae0c5474a8b323afac269055bf6ce8afa34](https://github.com/cockroachdb/cockroach/commits/59a69ae0c5474a8b323afac269055bf6ce8afa34):


```
=== RUN   TestTenantStreamingUnavailableStreamAddress
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantStreamingUnavailableStreamAddress2062442407
    test_log_scope.go:81: use -show-logs to present logs inline

pkg/ccl/streamingccl/replicationtestutils/testutils.go:149: ((*TenantStreamingClusters).init)
	NOTICE: .WaitForTenantCapabilities() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().WaitForTenantCapabilities() instead.
    testutils.go:487: condition failed to evaluate within 45s: from testutils.go:495: stream ingestion has not recorded any progress yet, waiting to advance pos 1697823898.371798248,0
    panic.go:522: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantStreamingUnavailableStreamAddress2062442407
--- FAIL: TestTenantStreamingUnavailableStreamAddress (284.77s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=3</code>
, <code>shard=9</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStreamingUnavailableStreamAddress.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/streamingccl/streamingest.TestTenantStreamingUnavailableStreamAddress [failed](https://tanzanite.cluster.engflow.com/invocation/817e8f2d-e4dc-4b32-bfbb-783ac3877f53) on master @ [59a69ae0c5474a8b323afac269055bf6ce8afa34](https://github.com/cockroachdb/cockroach/commits/59a69ae0c5474a8b323afac269055bf6ce8afa34):


```
=== RUN   TestTenantStreamingUnavailableStreamAddress
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantStreamingUnavailableStreamAddress2581567941
    test_log_scope.go:81: use -show-logs to present logs inline

pkg/ccl/streamingccl/replicationtestutils/testutils.go:149: ((*TenantStreamingClusters).init)
	NOTICE: .WaitForTenantCapabilities() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().WaitForTenantCapabilities() instead.
    testutils.go:487: condition failed to evaluate within 45s: from testutils.go:495: stream ingestion has not recorded any progress yet, waiting to advance pos 1697824013.168647417,0
    panic.go:522: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantStreamingUnavailableStreamAddress2581567941
--- FAIL: TestTenantStreamingUnavailableStreamAddress (391.93s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=4</code>
, <code>shard=9</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStreamingUnavailableStreamAddress.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/streamingccl/streamingest.TestTenantStreamingUnavailableStreamAddress [failed](https://tanzanite.cluster.engflow.com/invocation/817e8f2d-e4dc-4b32-bfbb-783ac3877f53) on master @ [59a69ae0c5474a8b323afac269055bf6ce8afa34](https://github.com/cockroachdb/cockroach/commits/59a69ae0c5474a8b323afac269055bf6ce8afa34):


```
=== RUN   TestTenantStreamingUnavailableStreamAddress
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantStreamingUnavailableStreamAddress2986234972
    test_log_scope.go:81: use -show-logs to present logs inline

pkg/ccl/streamingccl/replicationtestutils/testutils.go:149: ((*TenantStreamingClusters).init)
	NOTICE: .WaitForTenantCapabilities() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().WaitForTenantCapabilities() instead.
    testutils.go:487: condition failed to evaluate within 45s: from testutils.go:495: stream ingestion has not recorded any progress yet, waiting to advance pos 1697824880.237737584,0
    panic.go:522: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantStreamingUnavailableStreamAddress2986234972
--- FAIL: TestTenantStreamingUnavailableStreamAddress (398.50s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=19</code>
, <code>shard=9</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStreamingUnavailableStreamAddress.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/streamingccl/streamingest.TestTenantStreamingUnavailableStreamAddress [failed](https://tanzanite.cluster.engflow.com/invocation/817e8f2d-e4dc-4b32-bfbb-783ac3877f53) on master @ [59a69ae0c5474a8b323afac269055bf6ce8afa34](https://github.com/cockroachdb/cockroach/commits/59a69ae0c5474a8b323afac269055bf6ce8afa34):


```
=== RUN   TestTenantStreamingUnavailableStreamAddress
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantStreamingUnavailableStreamAddress4021452249
    test_log_scope.go:81: use -show-logs to present logs inline

pkg/ccl/streamingccl/replicationtestutils/testutils.go:149: ((*TenantStreamingClusters).init)
	NOTICE: .WaitForTenantCapabilities() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().WaitForTenantCapabilities() instead.
    testutils.go:487: condition failed to evaluate within 45s: from testutils.go:495: stream ingestion has not recorded any progress yet, waiting to advance pos 1697825073.844704636,0
    panic.go:522: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantStreamingUnavailableStreamAddress4021452249
--- FAIL: TestTenantStreamingUnavailableStreamAddress (383.57s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=22</code>
, <code>shard=9</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStreamingUnavailableStreamAddress.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/streamingccl/replicationtestutils/testutils.go
+++ b/pkg/ccl/streamingccl/replicationtestutils/testutils.go
@@ -166,11 +166,11 @@ func (c *TenantStreamingClusters) init(ctx context.Context) {
 // will not yet be active. If the caller passes withTestingKnobs, the
 // destination tenant starts up via a testServer.StartSharedProcessTenant().
 func (c *TenantStreamingClusters) StartDestTenant(
-	ctx context.Context, withTestingKnobs *base.TestingKnobs,
+	ctx context.Context, withTestingKnobs *base.TestingKnobs, server int,
 ) func() error {
 	if withTestingKnobs != nil {
 		var err error
-		_, c.DestTenantConn, err = c.DestCluster.Server(0).StartSharedProcessTenant(ctx, base.TestSharedProcessTenantArgs{
+		_, c.DestTenantConn, err = c.DestCluster.Server(server).StartSharedProcessTenant(ctx, base.TestSharedProcessTenantArgs{
 			TenantID:    c.Args.DestTenantID,
 			TenantName:  c.Args.DestTenantName,
 			Knobs:       *withTestingKnobs,
@@ -179,7 +179,7 @@ func (c *TenantStreamingClusters) StartDestTenant(
 		require.NoError(c.T, err)
 	} else {
 		c.DestSysSQL.Exec(c.T, `ALTER TENANT $1 START SERVICE SHARED`, c.Args.DestTenantName)
-		c.DestTenantConn = c.DestCluster.Server(0).SystemLayer().SQLConn(c.T, serverutils.DBName("cluster:"+string(c.Args.DestTenantName)+"/defaultdb"))
+		c.DestTenantConn = c.DestCluster.Server(server).SystemLayer().SQLConn(c.T, serverutils.DBName("cluster:"+string(c.Args.DestTenantName)+"/defaultdb"))
 	}
 
 	c.DestTenantSQL = sqlutils.MakeSQLRunner(c.DestTenantConn)
@@ -189,7 +189,7 @@ func (c *TenantStreamingClusters) StartDestTenant(
 	// TODO (msbutler): consider granting the new tenant some capabilities.
 	c.DestSysSQL.Exec(c.T, `ALTER TENANT $1 SET CLUSTER SETTING sql.virtual_cluster.feature_access.zone_configs.enabled=true`, c.Args.DestTenantName)
 	c.DestSysSQL.Exec(c.T, `ALTER TENANT $1 GRANT CAPABILITY can_use_nodelocal_storage`, c.Args.DestTenantName)
-	require.NoError(c.T, c.DestCluster.Server(0).WaitForTenantCapabilities(ctx, c.Args.DestTenantID, map[tenantcapabilities.ID]string{
+	require.NoError(c.T, c.DestCluster.Server(server).WaitForTenantCapabilities(ctx, c.Args.DestTenantID, map[tenantcapabilities.ID]string{
 		tenantcapabilities.CanUseNodelocalStorage: "true",
 	}, ""))
 	return func() error {

--- a/pkg/ccl/streamingccl/streamingest/alter_replication_job_test.go
+++ b/pkg/ccl/streamingccl/streamingest/alter_replication_job_test.go
@@ -114,7 +114,7 @@ func TestAlterTenantPauseResume(t *testing.T) {
 	cutoverOutput := replicationtestutils.DecimalTimeToHLC(t, cutoverStr)
 	require.Equal(t, cutoverTime, cutoverOutput.GoTime())
 	jobutils.WaitForJobToSucceed(c.T, c.DestSysSQL, jobspb.JobID(ingestionJobID))
-	cleanupTenant := c.StartDestTenant(ctx, nil)
+	cleanupTenant := c.StartDestTenant(ctx, nil, 0)
 	defer func() {
 		require.NoError(t, cleanupTenant())
 	}()

--- a/pkg/ccl/streamingccl/streamingest/datadriven_test.go
+++ b/pkg/ccl/streamingccl/streamingest/datadriven_test.go
@@ -155,7 +155,7 @@ func TestDataDriven(t *testing.T) {
 					jobspb.JobID(ds.ingestionJobID))
 			case "start-replicated-tenant":
 				testingKnobs := replicationtestutils.DefaultAppTenantTestingKnobs()
-				cleanupTenant := ds.replicationClusters.StartDestTenant(ctx, &testingKnobs)
+				cleanupTenant := ds.replicationClusters.StartDestTenant(ctx, &testingKnobs, 0)
 				ds.cleanupFns = append(ds.cleanupFns, cleanupTenant)
 			case "let":
 				if len(d.CmdArgs) == 0 {

--- a/pkg/ccl/streamingccl/streamingest/replication_stream_e2e_test.go
+++ b/pkg/ccl/streamingccl/streamingest/replication_stream_e2e_test.go
@@ -485,17 +485,16 @@ func TestTenantStreamingUnavailableStreamAddress(t *testing.T) {
 
 	skip.UnderDeadlock(t, "multi-node may time out under deadlock")
 	skip.UnderRace(t, "takes too long with multiple nodes")
+	skip.UnderStress(t, "multi node test times out under stress")
 
 	ctx := context.Background()
 	args := replicationtestutils.DefaultTenantStreamingClustersArgs
+	args.MultitenantSingleClusterNumNodes = 4
 
-	args.SrcNumNodes = 4
-	args.DestNumNodes = 3
-
-	c, cleanup := replicationtestutils.CreateTenantStreamingClusters(ctx, t, args)
+	c, cleanup := replicationtestutils.CreateMultiTenantStreamingCluster(ctx, t, args)
 	defer cleanup()
 
-	replicationtestutils.CreateScatteredTable(t, c, 3)
+	replicationtestutils.CreateScatteredTable(t, c, 4)
 	srcScatteredData := c.SrcTenantSQL.QueryStr(c.T, "SELECT * FROM d.scattered ORDER BY key")
 
 	producerJobID, ingestionJobID := c.StartStreamReplication(ctx)
@@ -526,6 +525,9 @@ func TestTenantStreamingUnavailableStreamAddress(t *testing.T) {
 	c.SrcTenantServer.AppStopper().Stop(ctx)
 	c.SrcCluster.StopServer(0)
 
+	// Switch the SQL connection to a new node, as node 0 has shutdown-- recall that
+	// the source and destination tenant are on the same cluster.
+	c.DestSysSQL = sqlutils.MakeSQLRunner(c.DestCluster.Conns[1])
 	c.DestSysSQL.Exec(t, `RESUME JOB $1`, ingestionJobID)
 	jobutils.WaitForJobToRun(t, c.DestSysSQL, jobspb.JobID(ingestionJobID))
 
@@ -537,7 +539,7 @@ func TestTenantStreamingUnavailableStreamAddress(t *testing.T) {
 	require.Equal(c.T, cutoverTime, cutoverOutput.GoTime())
 	jobutils.WaitForJobToSucceed(c.T, c.DestSysSQL, jobspb.JobID(ingestionJobID))
 
-	cleanUpTenant := c.StartDestTenant(ctx, nil)
+	cleanUpTenant := c.StartDestTenant(ctx, nil, 1)
 	defer func() {
 		require.NoError(t, cleanUpTenant())
 	}()
@@ -655,12 +657,11 @@ func TestTenantStreamingMultipleNodes(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	skip.UnderDeadlock(t, "multi-node may time out under deadlock")
-	skip.UnderRace(t, "takes too long with multiple nodes")
+	skip.UnderRace(t, "multi-node test may time out under race")
 
 	ctx := context.Background()
 	args := replicationtestutils.DefaultTenantStreamingClustersArgs
-	args.SrcNumNodes = 4
-	args.DestNumNodes = 3
+	args.MultitenantSingleClusterNumNodes = 3
 
 	// Track the number of unique addresses that were connected to
 	clientAddresses := make(map[string]struct{})
@@ -673,7 +674,7 @@ func TestTenantStreamingMultipleNodes(t *testing.T) {
 		},
 	}
 
-	c, cleanup := replicationtestutils.CreateTenantStreamingClusters(ctx, t, args)
+	c, cleanup := replicationtestutils.CreateMultiTenantStreamingCluster(ctx, t, args)
 	defer cleanup()
 
 	// Make sure we have data on all nodes, so that we will have multiple

